lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2246,6 +2247,7 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
     {
       llvm::raw_string_ostream stream(diffAttrReq);
       req->getAttrs().getAttribute<DifferentiableAttr>()->print(stream, req);
+      diffAttrReq = StringRef(stream.str()).trim();
```

Comment:
Note: I added logic to trim the printed attribute string. By default, an extra trailing space is always printed.